### ISSUE TITLE
exception handling in GetDifferences

### ISSUE THREAD

**Issue by klapantius** [replaysMike/AnyDiff#9]
I'm using AnyDiff to compare instances of the type `Microsoft.TeamFoundation.Build.WebApi.BuildDefinition` from assembly `Microsoft.TeamFoundation.Build2.WebApi (Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)`. In certain cases (I could not really understand why), the process breaks by a `System.NotSupportedException`:
 
```
System.NotSupportedException: Specified method is not supported.
   at System.Reflection.TypeInfo.<get_DeclaredNestedTypes>d__23.System.Collections.IEnumerator.Reset()
   at AnyDiff.DiffProvider.GetCountFromEnumerable(IEnumerable enumerable)
   at AnyDiff.DiffProvider.GetDifferences(String propertyName, Type propertyType, TypeConverter typeConverter, Object left, Object right, Object parent, ICollection``1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection``1 propertyList, DiffOptions diffOptions)
   at AnyDiff.DiffProvider.RecurseProperties(Object left, Object right, Object parent, ICollection``1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions comparisonOptions, ICollection``1 propertyList, DiffOptions diffOptions)
   at AnyDiff.DiffProvider.GetDifferences(String propertyName, Type propertyType, TypeConverter typeConverter, Object left, Object right, Object parent, ICollection``1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions options, ICollection``1 propertyList, DiffOptions diffOptions)
   at AnyDiff.DiffProvider.RecurseProperties(Object left, Object right, Object parent, ICollection``1 differences, Int32 currentDepth, Int32 maxDepth, ObjectHashcodeMap objectTree, String path, ComparisonOptions comparisonOptions, ICollection``1 propertyList, DiffOptions diffOptions)
...
```

I suggest to add a try/catch block to explain the `path` causing the not expected behavior, so that one can add it to the list of exceptions.

**Comment by replaysMike** [comment]
good suggestion, I'll try to repro and see if this needs to be handled or a bug in this check should be done in an alternative manner

**Comment by replaysMike** [comment]
so I've looked into this and it appears to be an odd implementation inside this library, however I've now taken this into consideration. While I did add a test for this and wasn't able to reproduce it - probably something to do with specific data being stored inside the `BuildDefinition`. But somewhere a collection is implementing the IEnumerable interface, however does not implement the Reset() method and throws a NotSupportedException. I am now accounting for this instead of executing it and assuming it would always work.

Should be fixed as of version 1.0.71
